### PR TITLE
ComboBox supports Dim.Fill() Dim.Percent()

### PR DESCRIPTION
**Pull request by fergusonr**

### CODE REVIEW

File: Terminal.Gui/Views/ComboBox.cs [hunk 1]
```diff
@@ -40,8 +40,8 @@ public class ComboBox : View {
 		/// </summary>
 		public ComboBox () : base()
 		{
-			search = new TextField ("") { LayoutStyle = LayoutStyle.Computed };
-			listview = new ListView () { LayoutStyle = LayoutStyle.Computed, CanFocus = true /* why? */ };
+			search = new TextField ("");
+			listview = new ListView () { LayoutStyle = LayoutStyle.Computed, CanFocus = true };
```
Comment (tig) [inline Terminal.Gui/Views/ComboBox.cs:44]: I noticed you do not expose the `Source` Property of `ListView`.

Won't users expect a combobox to have the same API as `Listview` when dealing with the list source?

File: Terminal.Gui/Core/PosDim.cs [hunk 2]
```diff
@@ -361,7 +361,7 @@ internal virtual int Anchor (int width)
 			return 0;
 		}
 
-		class DimFactor : Dim {
+		internal class DimFactor : Dim {
```
Comment (tig) [inline Terminal.Gui/Core/PosDim.cs:364]: This is the first class where access to these internals have appeared to be needed. This makes me very nervous as @migueldeicaza was very careful in the design. 

In all of my work on `Terminal.Gui` I've been able to find out ways to make things like what you are trying here work without accessing these internals.

What is it about ComboBox that requires this change?

File: Terminal.Gui/Views/ComboBox.cs [hunk 3]
```diff
@@ -81,7 +81,7 @@ private void Initialize()
 			};
 
 			// TODO: LayoutComplete event breaks cursor up/down. Revert to Application.Loaded
```
Comment (tig) [inline Terminal.Gui/Views/ComboBox.cs:83]: I just noticed this comment. What about `LayoutComplete` breaks cursor up/down? Have you submitted a bug against that? It shouldn't.

File: Terminal.Gui/Views/ComboBox.cs [hunk 4]
```diff
@@ -81,7 +81,7 @@ private void Initialize()
 			};
 
 			// TODO: LayoutComplete event breaks cursor up/down. Revert to Application.Loaded 
-			Application.Loaded += (sender, a) => {
+			Application.Loaded += (object sender, Application.ResizedEventArgs a) => {
```
Comment (tig) [inline Terminal.Gui/Views/ComboBox.cs:84]: This won't work. To prove it change your demo code in `UI Catalog` to:

```csharp
// ComboBox
var lbComboBox = new Label ("ComboBox") {
	ColorScheme = Colors.TopLevel,
	X = Pos.Right (lbListView) + 1,
	Width = Dim.Percent(30)
};

var comboBox = new ComboBox () {
	X = Pos.Right (listview) + 1,
	Y = Pos.Bottom (lbListView) + 1,
	Height = Dim.Fill (2),
	Width = Dim.Percent (30)
};
```

You'll see the combo box does not size with the parent.

**Comment by tig** [comment]
Can you please rebase?

**Comment by fergusonr** [comment]
Rebase done. I still need to be the resize Height implemented

**Comment by tig** [comment]
I'm still showing 104 files changed in this PR. I can't merge that.

**Comment by BDisp** [comment]
@fergusonr may you have to force the rebase.

**Comment by fergusonr** [comment]
Apologies, I'm relatively new to git. I managed to do a fresh clone of master and merge ok myself. What exactly is the command line for a "rebase force"? 

thanks

image

**Comment by BDisp** [comment]
I do all operations with the `vs2019 GitHub` extension. But you can do something like that:

`git clone https://github.com/migueldeicaza/gui.cs.git`
But this is not need as you already did it.

```
git remote add upstream https://github.com/migueldeicaza/gui.cs.git
git checkout master
git pull upstream/master
git rebase master
git push
```

If you have many conflicts here try to do a force push.
`git push -f`
This above allows you have your `master `branch even with the `upstream/master` branch.

```
git checkout combobox-fixes
git rebase master

```
If you have conflicts here and you want preserve your code modifications you must resolve otherwise you may loose the changes.

```
git commit blabla
git push -f
```
Here you can use force to force your `origin/combobox-fixes` to update with your local branch.

I wope I didn't say anything wrong but you can also search in google for help too.

**Comment by fergusonr** [comment]
OK thanks, I think that's done

**Comment by BDisp** [comment]
You have to add the ViewCombo.cs and ListAndCombo.cs to the projects first.

**Comment by fergusonr** [comment]
Crash is fixed. Investigating other issue....

**Comment by tig** [comment]
Thanks.

**Pull request merged**